src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -88,48 +86,62 @@ internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
 
         private void WriteLocations(Location location, IReadOnlyList<Location> additionalLocations)
         {
-            _writer.WriteArrayStart("locations");
+            if (location.SourceTree != null)
+            {
+                _writer.WriteArrayStart("locations");
+                _writer.WriteObjectStart(); // location
+                _writer.WriteKey("analysisTarget");
```

Comment:
Per the spec, analysisTarget is the file the tool was instructed to scan. resultFile is the file where the result was detected, if different from analysisTarget. The spec gives two examples:

1. A C++ analysis tool is instructed to scan foo.cpp (analysisTarget), but foo.cpp #include's bar.h, and the tool detects a result in bar.h (resultFile).
2. A binary analysis tool is instructed to scan foo.dll (analysisTarget), but the tool has symbol information telling it that the result it just detected occurs in source file bar.cs (resultFile).

The Roslyn scenario doesn't exactly match either of these examples. Roslyn was instructed to *compile* a.cs, ..., z.cs to *produce* foo.dll:

csc a.cs ... b.cs /out:foo.dll

Which files were the analyzers "instructed to *scan*"? Both the source files and the binary file are on the command line. But analyzers scan source files (they build up syntax trees, after all), so in my opinion analysisTarget is more appropriate here. 

---
In reply to: [62950732](https://github.com/dotnet/roslyn/pull/11233#discussion_r62950732) [](ancestors = 62950732)